tests/integration/golang/aim/run/search_test.go

```diff
@@ -469,12 +469,17 @@ func (s *SearchTestSuite) TestStreamData_Ok() {
 			},
 		},
 		{
-			name: "NoExperimentNamesNoResults",
+			name: "NoExperimentNamesQueryResults",
 			request: request.SearchRunsRequest{
 				Query:           `run.archived == True or run.archived == False`,
 				ExperimentNames: []string{},
 			},
-			runs: []*models.Run{},
+			runs: []*models.Run{
+				s.run1,
+				s.run2,
+				s.run3,
+				s.run4,
+			},
```

Comment:
this shows the problem -- there could be 1500 runs per experiment, and this would return them all!